Stop converting from LayoutUnit to Length in RenderBox computeLogical{Width, Height} functions .

sammygill opened the pull request:
#### 8a7e32536bc5938e07a1d701e3cf59a4c1204ac3
<pre>
Stop converting from LayoutUnit to Length in RenderBox computeLogical{Width, Height} functions .
<a href="https://bugs.webkit.org/show_bug.cgi?id=301232">https://bugs.webkit.org/show_bug.cgi?id=301232</a>
<a href="https://rdar.apple.com/problem/163150604">rdar://problem/163150604</a>

Reviewed by Alan Baradlay.

In preparation for being able to zoom the Preferred, Minimum, and Maximum
sizes, we need to stop the conversion of LayoutUnits to Lengths in parts
of RenderBox::computeLogicalWidth and RenderBox::computeLogicalHeight.
This is because these values should already be zoomed and doing this
conversion would result in double zooming when we evaluate the length
that we create.

The computedHeightValue seems to do a mix of checking if it has a used
logical height from its formatting context, can compute a used logical
height from the aspect ratio, or falling back to its computed logical
height from style. In the cases where it can determine a used logical
height, checkMinMaxHeight is false so we end up returning this fixed
length value. Otherwise, it falls back to the logical height it got from
its RenderStyle or possibly the overriding one it’s using for its flex
basis computation.

I separated out these two pieces of logic into their own lambdas. The
usedLogicalHeightFromContext lambda returns an optional LayoutUnit to
indicate whether or not it was able to determine one matching the
checkMinMaxHeight logic.

Canonical link: <a href="https://commits.webkit.org/301969@main">https://commits.webkit.org/301969@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8d71db37547be98d604a4f45bdddcd65c6c5c952

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127492 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47140 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38278 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134568 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/79048 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ff56919d-40ca-4b85-914f-54445be638d5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/129364 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47757 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55664 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/97043 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64988 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/38f573db-26f7-4a70-b05b-4f8c7e6b9fa4) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130440 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38199 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114181 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77524 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ef184171-1148-492b-b0e7-f9b18ceabe67) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/37027 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/32285 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77942 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/108059 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32709 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/137053 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/54151 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41727 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105571 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54662 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110537 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105222 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26861 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50737 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/29190 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51730 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/54088 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/60175 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53322 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56779 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/55081 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay reviewed (approved):
This is a bit hard_to_follow diff but rs=me.

sammygill committed:
Stop converting from LayoutUnit to Length in RenderBox computeLogical{Width, Height} functions .
https://bugs.webkit.org/show_bug.cgi?id=301232
rdar://problem/163150604

Reviewed by Alan Baradlay.

In preparation for being able to zoom the Preferred, Minimum, and Maximum
sizes, we need to stop the conversion of LayoutUnits to Lengths in parts
of RenderBox::computeLogicalWidth and RenderBox::computeLogicalHeight.
This is because these values should already be zoomed and doing this
conversion would result in double zooming when we evaluate the length
that we create.

The computedHeightValue seems to do a mix of checking if it has a used
logical height from its formatting context, can compute a used logical
height from the aspect ratio, or falling back to its computed logical
height from style. In the cases where it can determine a used logical
height, checkMinMaxHeight is false so we end up returning this fixed
length value. Otherwise, it falls back to the logical height it got from
its RenderStyle or possibly the overriding one it’s using for its flex
basis computation.

I separated out these two pieces of logic into their own lambdas. The
usedLogicalHeightFromContext lambda returns an optional LayoutUnit to
indicate whether or not it was able to determine one matching the
checkMinMaxHeight logic.

Canonical link: https://commits.webkit.org/301969@main
